### ISSUE TITLE
Question - Data from memory to display

### ISSUE THREAD

**Issue by gnkarn** [FWeinb/ESP8266-WebSocket-Demo#2]
Hi, you have a very good project here , what interested me is the functionality to receive Bin data via websockets and send it to the display , at high fps.

im trying to do something similar but instead to a display , to an led matrix .

My question , is after you have the data on your webimage memory , where is the code that is reading that memory segment and drawing to the 1306?
i don't see any on the loop ,

second question, how do you control , the frame rate that you are sending , in order to avoid overflowing the websocket on the ESP side ?
thank you

**Comment by tobozo** [comment]
Hi,

Found the answer to your first question by searching "webImage" in this project : https://github.com/FWeinb/ESP8266-WebSocket-Demo/blob/b38c079efa252ae52357e307f06fe60071c5b6af/src/DrawFunctions.h#L28

Search for "tick" and you'll find the answer to your second question.

**Comment by gnkarn** [comment]
thank you , but there should be a call to webFrame somewhere but there is no call to it on any of the source files ..
what am i missing ?

**Comment by tobozo** [comment]
this is where the binary data (payload) is received by the websocket and copied to webimage: https://github.com/FWeinb/ESP8266-WebSocket-Demo/blob/b38c079efa252ae52357e307f06fe60071c5b6af/src/WebSocket.h#L5

**Comment by gnkarn** [comment]
Yes I got that part , the part missing is where the pimage go to the screen .
There is no Call to any función , that is hidden somewhere , is it interrupt driven so is not in the loop?

**Comment by tobozo** [comment]
That was linked in my first answer, and it's a callback, which means it's triggered by an event (e.g. websocket received bin data), so you won't see a proper cast of this in the code, only declarations.
